tests/test_downloadermiddleware_retry.py

```diff
@@ -103,6 +103,34 @@ def _test_retry_exception(self, req, exception):
         req = self.mw.process_exception(req, exception, self.spider)
         self.assertEqual(req, None)
 
+    def test_different_retry(self):
+        
+        req = Request('http://www.scrapytest.org/invalid_url', meta={'max_retry_times': 1})
+        self._test_retry(req, DNSLookupError('foo'))
+        req2 = Request('http://www.scrapytest.org/invalid_url')
+        self._test_retry(req2, DNSLookupError('foo'))
+
+        stats = self.crawler.stats
+        assert stats.get_value('retry/max_reached') == 2
+        assert stats.get_value('retry/count') == 3
+
+    def _test_retry(self, req, exception):
+        
+        req = self.mw.process_exception(req, exception, self.spider)
+        assert isinstance(req, Request)
+
+        retry_times = req.meta.get('max_retry_times') or self.mw.max_retry_times
+
+        while req.meta['retry_times'] != retry_times:
+            req = self.mw.process_exception(req, exception, self.spider)
+            assert isinstance(req, Request)
+        
+        self.assertEqual(req.meta['retry_times'], retry_times)
```

Comment:
Can I suggest that `_test_retry` accepts an argument to actually test an expected retry times number?
so that `retry_times` variable in the test does not mimic what the implementation in the middleware does,
i.e. you're testing a behavior:

> if I set a max retry number in request meta, is the request retried X many times?,

instead of
> I know how retry time is computed, so I'm testing that the same code gives the same result

Also, I would find interesting to test with different `RETRY_TIMES` settings: `RETRY_TIMES=0`, `RETRY_TIMES` bigger, than `meta['max_retry_times']`, lower than etc.